chia/full_node/mempool_manager.py

```diff
@@ -696,12 +713,23 @@ async def new_peak(
             # find the same transaction multiple times. We put them in a set
             # to deduplicate
             spendbundle_ids_to_remove: Set[bytes32] = set()
+            # Gather also the singleton fast forward puzzle hashes to be removed
+            # from the unspent lineage info map
+            ff_puzzle_hashes_to_remove: Set[bytes32] = set()
             for spend in spent_coins:
                 items: List[MempoolItem] = self.mempool.get_items_by_coin_id(spend)
                 for item in items:
                     included_items.append(MempoolItemInfo(item.cost, item.fee, item.height_added_to_mempool))
                     self.remove_seen(item.name)
                     spendbundle_ids_to_remove.add(item.name)
+                    ff_puzzle_hashes_to_remove = {
+                        spend_data.coin_spend.coin.puzzle_hash
+                        for spend_data in item.bundle_coin_spends.values()
+                        if spend_data.eligible_for_fast_forward
+                    }
+            self.puzzle_hash_to_unspent_lineage_info = {
```

Comment:
this looks like a slow way of removing items, but maybe it's the way it's done.
I would think mutating `self.puzzle_hash_to_unspent_lineage_info` would be faster